### PR TITLE
Replaced WebL10n with i18next in Markdown activity

### PR DESCRIPTION
**Pull request by soham2312**
fix issue #1378 

Replaced Code in MarkDown Activity to l10n

@llaske  , could you please review it

### CODE REVIEW

**Comment by llaske** [comment]
Could you avoid to change file formatting.
Only updated lines must be changed in files. It's important for maintenance reason.
I guess it's coming from your editor settings.

**Comment by soham2312** [comment]
Sure @llaske

**Comment by soham2312** [comment]
I have tried to make the changes as much as possible @llaske please review it

File: activities/Markdown.activity/js/activity.js [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
 
 	// Manipulate the DOM only when it is ready.
 	requirejs(['domReady!'], function (doc) {
 
 		// Initialize the activity.
+		l10n = _l10n
 		activity.setup();
 		var buttons = ["insertText", "wmd-bold-button-second", "wmd-italic-button-second", "wmd-heading-button", "wmd-hr-button",
-				"wmd-olist-button", "wmd-ulist-button", "wmd-code-button", "wmd-quote-button", "wmd-link-button",
-				"wmd-undo-button", "wmd-redo-button", "wmd-showHideEditor-button","wmd-showHidePreview-button"];
```
Comment (llaske) [inline activities/Markdown.activity/js/activity.js:10]: Indentation change

File: activities/Markdown.activity/js/activity.js [hunk 2]
```diff
@@ -18,7 +19,7 @@ define(["sugar-web/activity/activity","webL10n","sugar-web/datastore"], function
 
 			var jsonData = JSON.stringify((inputTextContent.value).toString());
 			datastoreObject.setDataAsText(jsonData);
-			datastoreObject.save(function () {});
+			datastoreObject.save(function () { });
```
Comment (llaske) [inline activities/Markdown.activity/js/activity.js:22]: Indentation change

File: activities/Markdown.activity/js/activity.js [hunk 3]
```diff
@@ -56,39 +57,35 @@ define(["sugar-web/activity/activity","webL10n","sugar-web/datastore"], function
 		var preview = document.getElementById("wmd-preview-second");
 		var textArea = document.getElementById("wmd-input-second");
 
-		function isElementHidden (element) {
+		function isElementHidden(element) {
```
Comment (llaske) [inline activities/Markdown.activity/js/activity.js:60]: Indentation change

File: activities/Markdown.activity/js/activity.js [hunk 4]
```diff
@@ -56,39 +57,35 @@ define(["sugar-web/activity/activity","webL10n","sugar-web/datastore"], function
...
 			return window.getComputedStyle(element, null).getPropertyValue('display') === 'none';
 		}
-		showHideEditor.onclick = function(){
+		showHideEditor.onclick = function () {
```
Comment (llaske) [inline activities/Markdown.activity/js/activity.js:63]: Indentation change

File: activities/Markdown.activity/js/activity.js [hunk 5]
```diff
@@ -56,39 +57,35 @@ define(["sugar-web/activity/activity","webL10n","sugar-web/datastore"], function
...
 
-			if(isElementHidden(preview))
-			{
+			if (isElementHidden(preview)) {
```
Comment (llaske) [inline activities/Markdown.activity/js/activity.js:65]: Identation change

File: activities/Markdown.activity/js/activity.js [hunk 6]
```diff
@@ -56,39 +57,35 @@ define(["sugar-web/activity/activity","webL10n","sugar-web/datastore"], function
...
 				preview.style.display = "inline";
 				preview.style.width = "97%";
 				panel.style.width = "1%";
 			}
-			if(isElementHidden(panel))
-			{
+			if (isElementHidden(panel)) {
```
Comment (llaske) [inline activities/Markdown.activity/js/activity.js:70]: Identation change

File: activities/Markdown.activity/js/activity.js [hunk 7]
```diff
@@ -56,39 +57,35 @@ define(["sugar-web/activity/activity","webL10n","sugar-web/datastore"], function
...
 				panel.style.display = "inline";
 				panel.style.width = "47%";
 				preview.style.width = "47%";
 			}
-			else{
+			else {
```
Comment (llaske) [inline activities/Markdown.activity/js/activity.js:75]: Identation change

File: activities/Markdown.activity/js/activity.js [hunk 8]
```diff
@@ -56,39 +57,35 @@ define(["sugar-web/activity/activity","webL10n","sugar-web/datastore"], function
...
 				panel.style.display = "none";
 				panel.style.width = "1%";
 				preview.style.width = "97%";
 			}
 
 		}
-		showHidePreview.onclick = function(){
-			if(isElementHidden(panel))
-			{
+		showHidePreview.onclick = function () {
```
Comment (llaske) [inline activities/Markdown.activity/js/activity.js:82]: Identation change

File: activities/Markdown.activity/js/activity.js [hunk 9]
```diff
@@ -56,39 +57,35 @@ define(["sugar-web/activity/activity","webL10n","sugar-web/datastore"], function
...
+			if (isElementHidden(panel)) {
 				panel.style.display = "inline";
 				panel.style.width = "97%";
 				preview.style.width = "1%";
 			}
-			if(isElementHidden(preview))
-			{
+			if (isElementHidden(preview)) {
```
Comment (llaske) [inline activities/Markdown.activity/js/activity.js:88]: Identation change

**Review by llaske — CHANGES_REQUESTED** [review]
You still have a lot of indentation change only.
Plus you've got the same issue than this one https://github.com/llaske/sugarizer/pull/1533#issuecomment-1955152552

**Comment by soham2312** [comment]
@llaske Resolved all the indentations mentioned

File: activities/Markdown.activity/js/activity.js [hunk 10]
```diff
@@ -56,38 +57,38 @@ define(["sugar-web/activity/activity","webL10n","sugar-web/datastore"], function
 		var preview = document.getElementById("wmd-preview-second");
 		var textArea = document.getElementById("wmd-input-second");
 
-		function isElementHidden (element) {
+		function isElementHidden(element){
```
Comment (llaske) [inline activities/Markdown.activity/js/activity.js:60]: Still indentation issue.

File: activities/Markdown.activity/js/activity.js [hunk 11]
```diff
@@ -56,38 +57,38 @@ define(["sugar-web/activity/activity","webL10n","sugar-web/datastore"], function
...
 			return window.getComputedStyle(element, null).getPropertyValue('display') === 'none';
 		}
 		showHideEditor.onclick = function(){
 
-			if(isElementHidden(preview))
+			if (isElementHidden(preview))
```
Comment (llaske) [inline activities/Markdown.activity/js/activity.js:65]: Still indentation issue.

File: activities/Markdown.activity/js/activity.js [hunk 12]
```diff
@@ -56,38 +57,38 @@ define(["sugar-web/activity/activity","webL10n","sugar-web/datastore"], function
...
 			{
 				preview.style.display = "inline";
 				preview.style.width = "97%";
 				panel.style.width = "1%";
 			}
-			if(isElementHidden(panel))
+			if (isElementHidden(panel))
```
Comment (llaske) [inline activities/Markdown.activity/js/activity.js:71]: Still indentation issue.

File: activities/Markdown.activity/js/activity.js [hunk 13]
```diff
@@ -56,38 +57,38 @@ define(["sugar-web/activity/activity","webL10n","sugar-web/datastore"], function
...
 			{
 				panel.style.display = "inline";
 				panel.style.width = "47%";
 				preview.style.width = "47%";
 			}
-			else{
+			else {
```
Comment (llaske) [inline activities/Markdown.activity/js/activity.js:77]: Still indentation issue.

File: activities/Markdown.activity/js/activity.js [hunk 14]
```diff
@@ -56,38 +57,38 @@ define(["sugar-web/activity/activity","webL10n","sugar-web/datastore"], function
...
 				panel.style.display = "none";
 				panel.style.width = "1%";
 				preview.style.width = "97%";
 			}
 
 		}
-		showHidePreview.onclick = function(){
-			if(isElementHidden(panel))
+		showHidePreview.onclick = function () {
```
Comment (llaske) [inline activities/Markdown.activity/js/activity.js:84]: Still indentation issue.

File: activities/Markdown.activity/js/activity.js [hunk 15]
```diff
@@ -56,38 +57,38 @@ define(["sugar-web/activity/activity","webL10n","sugar-web/datastore"], function
...
+			if (isElementHidden(panel))
 			{
 				panel.style.display = "inline";
 				panel.style.width = "97%";
 				preview.style.width = "1%";
 			}
-			if(isElementHidden(preview))
+			if (isElementHidden(preview))
```
Comment (llaske) [inline activities/Markdown.activity/js/activity.js:91]: Still indentation issue.

**Review by llaske — CHANGES_REQUESTED** [review]
Still indentation change: only lines changed related to the issue must be updated.
Plus the issue regarding fallback language loading should be solved (see https://github.com/llaske/sugarizer/pull/1533#issuecomment-1955152552)

**Pull request merged**